src/mirco_nonlinearsolver.cpp

```diff
@@ -149,8 +149,7 @@ void MIRCO::NonLinearSolver::NonlinearSolve(const Teuchos::SerialDenseMatrix<int
         }
       }
       // Solving solverMatrix*vector_x=vector_b
-      MIRCO::LinearSolver solution;
-      solution.Solve(solverMatrix, vector_x, vector_b);
+      MIRCO::LinearSolver::Solve(solverMatrix, vector_x, vector_b);
```

Comment:
Would it make sense to return `vector_x` (instead of passing it as a reference? Matrix and right-hand side should be const in the linear solver, and the solution should depend on those two only, right?

Reply:
Yes, that is what my understanding it as well. I am unaware what difference does it make to return a variable instead if passing it as a reference to edit it.